src/App.tsx

```diff
@@ -1,61 +1,103 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import TodoList from './components/TodoList/TodoList';
+import { User } from './Types/User';
+import { Todo } from './Types/Todo';
+import { getNewTodoId } from './components/utils/utils';
+
+const initialTodos = todosFromServer;
+const initialUsers = usersFromServer;
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+  const users: User[] = initialUsers;
+  const [isTitleInvalid, setIsTitleInvalid] = useState(false);
+  const [isUserValid, setIsUserValid] = useState(false);
+  const [newTitle, setNewTitle] = useState('');
+  const [newUser, setNewUser] = useState(0);
+
+  const validFormUser = () => (newUser
+    ? (setIsUserValid(false), false) : (setIsUserValid(true), true));
+
+  const validFormTitle = () => (newTitle
+    ? (setIsTitleInvalid(false), false) : (setIsTitleInvalid(true), true));
+
+  const addTodo = () => {
+    if (!validFormTitle() && !validFormUser()) {
```

Comment:
So title and user should be invalid to submit the form? The names are a bit confusing